wp-includes/service-workers.php

```diff
@@ -236,4 +236,14 @@ function wp_default_service_workers( $service_workers ) {
 				);
 		}
 	}
+
+	add_action( 'wp_admin_service_worker', array( $service_workers, 'precache_admin_assets' ), 9 );
+}
+
+/**
+ * Disables concatenating scripts to leverage caching the assets via Service Worker instead.
+ */
+function wp_disable_script_concatenation() {
```

Comment:
Something we need to look into is what to do for users who don't have service workers available. In their case, we actually need to retain the concatenation functionality.

The only option that comes to mind is that when the promise [returned](https://github.com/xwp/pwa-wp/blob/57ebfa7652cd6e485056c7d11417acb726a7e87d/wp-includes/service-workers.php#L110-L117) by `navigator.serviceWorker.register()` resolves:

```diff
diff --git a/wp-includes/service-workers.php b/wp-includes/service-workers.php
index 69a39fb..4d2029c 100644
--- a/wp-includes/service-workers.php
+++ b/wp-includes/service-workers.php
@@ -112,7 +112,9 @@ function wp_print_service_workers() {
 					navigator.serviceWorker.register(
 						<?php echo wp_json_encode( wp_get_service_worker_url( $name ) ); ?>,
 						<?php echo wp_json_encode( compact( 'scope' ) ); ?>
-					);
+					).then( function() {
+						document.cookie = 'wordpress_sw_installed=1; path=/; expires=Fri, 31 Dec 9999 23:59:59 GMT; secure; samesite=strict';
+					} );
 				<?php } ?>
 			} );
 		}
```

That we can set a cookie like `wordpress_sw_installed` which can then be used to check for whether to proceed with concatenation. 

So then in this `wp_disable_script_concatenation` it can first check if `isset( $_COOKIE['wordpress_sw_installed'] )` before proceeding.

Note that concatenation only applies in the admin, which entails that the user is authenticated, and no full-page caching is used. Thus there won't be concerns about reading the cookie since these pages won't be cached.

Reply:
@westonruter Thanks for bringing that issue out and also for the solution suggestion, good idea!

Added within ec569c4.